src/common/lock.c

```diff
@@ -261,7 +296,9 @@ lockRelease(bool failOnNoLock)
 
         for (LockType lockIdx = lockMin; lockIdx <= lockMax; lockIdx++)
         {
-            lockReleaseFile(lockFd[lockIdx], lockFile[lockIdx]);
+            if (lockFd[lockIdx] != LOCK_ON_EXEC_ID)
+                lockReleaseFile(lockFd[lockIdx], lockFile[lockIdx]);
```

Comment:
Sorry, I approved the commit but had another couple of questions:
1. When are we ever releasing the file that we said was locked because of LOCK_ON_EXEC_ID? We are performing a strFree on the lockFile but may not be performing a lockRelease on it yet. 
2. In command/info/info.c we attempt to get a lock on a file and then we release it see https://github.com/pgbackrest/pgbackrest/blob/master/src/command/info/info.c#L468 so how willt his be affected?

Reply:
1. We can't free the lock (and remove the file) because we don't hold the actual lock. The remote that acquired the lock will need to free it.
2. It's not affected because the info process will have a different exec-id so won't be able to share the lock if it already exists. This is tested at line 220 in `command/infoTest.c`.